File: file-main-backup/fix_imports.js
const fs = require('fs');
const path = require('path');

function processDirectory(directory) {
    const pattern = /^(\s*)import\s+(\w+)\s+from\s+['"]figma:asset\/[^'"]+['"];?\s*$/gm;
    let count = 0;

    function walkSync(currentDirPath) {
        fs.readdirSync(currentDirPath).forEach(function (name) {
            var filePath = path.join(currentDirPath, name);
            var stat = fs.statSync(filePath);
            if (stat.isFile() && /\.(tsx|ts|jsx|js)$/.test(filePath)) {
                let content = fs.readFileSync(filePath, 'utf8');
                let newContent = content.replace(pattern, (match, indent, varName) => {
                    let pathName = varName;
                    if (pathName.startsWith('imgImage')) {
                        pathName = pathName.substring(8);
                    } else if (pathName.startsWith('img')) {
                        pathName = pathName.substring(3);
                    }
                    if (pathName) {
                        pathName = pathName.charAt(0).toLowerCase() + pathName.slice(1);
                    } else {
                        pathName = "asset";
                    }
                    return `${indent}const ${varName} = "/${pathName}.png";`;
                });
                if (newContent !== content) {
                    fs.writeFileSync(filePath, newContent, 'utf8');
                    console.log(`Updated ${filePath}`);
                    count++;
                }
            } else if (stat.isDirectory() && name !== 'node_modules') {
                walkSync(filePath);
            }
        });
    }

    walkSync(directory);
    console.log(`Total files updated: ${count}`);
}

processDirectory("c:/Users/1sriv/Downloads/file-main/file-main/src");

File: file-main-backup/fix_imports.py
import os
import re

def process_directory(directory):
    # Regex to match import line: import imgName from "figma:asset/..."
    pattern = re.compile(r'^(\s*)import\s+(\w+)\s+from\s+[\'"]figma:asset/[^\'"]+[\'"];?\s*$', re.MULTILINE)
    
    count = 0
    for root, dirs, files in os.walk(directory):
        for file in files:
            if not file.endswith(('.tsx', '.ts', '.jsx', '.js')):
                continue
            
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
                
            def replacer(match):
                indent = match.group(1)
                var_name = match.group(2)
                
                path_name = var_name
                if path_name.startswith('imgImage'):
                    path_name = path_name[8:]
                elif path_name.startswith('img'):
                    path_name = path_name[3:]
                
                if path_name:
                    path_name = path_name[0].lower() + path_name[1:]
                else:
                    path_name = "asset"
                    
                return f'{indent}const {var_name} = "/{path_name}.png";'
                
            new_content = pattern.sub(replacer, content)
            
            if new_content != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {filepath}")
                count += 1
                
    print(f"Total files updated: {count}")

if __name__ == "__main__":
    process_directory(r"c:\Users\1sriv\Downloads\file-main\file-main\src")
